src/libmongoc/tests/test-mongoc-streamable-hello.c

```diff
@@ -52,7 +52,7 @@ test_topology_version_update (void)
    mock_server_run (server);
 
    client =
-      test_framework_client_new_from_uri (mock_server_get_uri (server), NULL);
+      mongoc_client_new_from_uri (mock_server_get_uri (server));
```

Comment:
Please update this file to use test_framework_client_pool_new_from_uri and replace the calls to mock_server_receives_legacy_hello with mock_server_receives_any_hello.